packages/engine-test-utils/src/__tests__/engine-server/EngineEvents.spec.ts

```diff
@@ -155,19 +153,52 @@ describe("GIVEN a DendronEngineClient running on client-side", () => {
 
           engineClient.onEngineNoteStateChanged(
             (noteChangeEntries: NoteChangeEntry[]) => {
-              try {
-                noteChangeEntries.forEach((entry) => {
-                  // TODO: Add validation once scenario works.
-                  console.log(entry);
+              const createEntries = extractNoteChangeEntriesByType(
+                noteChangeEntries,
+                "create"
+              );
+
+              const deleteEntries = extractNoteChangeEntriesByType(
+                noteChangeEntries,
+                "delete"
+              );
+
+              const updateEntries = extractNoteChangeEntriesByType(
+                noteChangeEntries,
+                "update"
+              ) as NoteChangeUpdateEntry[];
+
+              testAssertsInsideCallback(() => {
+                expect(createEntries.length).toEqual(1);
+                expect(updateEntries.length).toEqual(2);
+                expect(deleteEntries.length).toEqual(1);
+
+                expect(createEntries[0].note.fname).toEqual("foo");
+                expect(createEntries[0].note.id).toEqual("updatedID");
+                expect(deleteEntries[0].note.fname).toEqual("foo");
+                expect(deleteEntries[0].note.id).toEqual("foo");
+
+                updateEntries.forEach((entry) => {
+                  if (entry.note.fname === "root") {
+                    expect(entry.note.children).toEqual(["bar", "updatedID"]);
+                    expect(entry.status).toEqual("update");
+                  } else if (entry.note.fname === "foo.ch1") {
+                    expect(entry.note.parent).toEqual("updatedID");
+                    expect(entry.status).toEqual("update");
+                  } else {
+                    done({
+                      message: `Unexpected NoteChangeEntry. fname: ${entry.note.fname}, id: ${entry.note.id}, status: ${entry.status}`,
+                    });
+                    return;
+                  }
                 });
+
                 done();
-              } catch (err) {
-                done(err);
-              }
+              }, done);
             }
           );
 
-          await engineClient.writeNote(fooUpdated, { updateExisting: true });
+          await engineClient.writeNote(fooUpdated);
```

Comment:
It's still not very clear to me on whether the `{updateExisting: true}` flag is necessary to be set here or not to trigger this behavior. 

IMO - this _should_ be the correct behavior regardless of whether this flag is set or not - @kevinslin thoughts?

Reply:
I don't think for this case setting `updateExisting: true` is right. we want to overwrite.

in `engine.writeNote` if `updateExisting: true` is set, we don't go into the logic at all where the overwrite happens.

Reply:
for anything other than changing the note's id, `updateExisting: true` makes sense.